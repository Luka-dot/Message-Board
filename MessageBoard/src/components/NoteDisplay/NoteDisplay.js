import React from 'react';

const NoteDisplay = (props) => {
  const titleMax = 20;
  const listMax = 30;
  return (
    <div className="NoteDisplay">
      <h3>
        {
          props.title.length > titleMax ?
          props.title.substring(0, titleMax) + '...' :
          props.title
        }
      </h3>
      <ul>
      {
        props.list.map(item => (
          <li key={item.id} className={item.isComplete ? 'completed' : ''}>
            {
              item.text.length > listMax ?
              item.text.substring(0, listMax) + '...' :
              item.text
            }
          </li>
        ))
      }
      </ul>
    </div>
  )

}

export default NoteDisplay;