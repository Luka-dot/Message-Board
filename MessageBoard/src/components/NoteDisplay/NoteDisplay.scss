.NoteDisplay {
  display: block;
  pointer-events: none;

  h3 {
    color: $textColor;
    margin: 0;
    text-align: left;
    padding-bottom: .2em;
  }

  ul {
    text-align: left;
    padding: 0;
    margin: 0;

    li {
      color: $textColor;
      list-style-type: none;
      display: flex;
      justify-content: flex-start;
      align-items: flex-start;
      line-height: 1.2;
      padding-bottom: .1em;

      .checkBox {
        opacity: .1;
        img {
          max-width: 75%;
        }
      }

    }

    li.completed {
      text-decoration: line-through;
    }

  }

}
