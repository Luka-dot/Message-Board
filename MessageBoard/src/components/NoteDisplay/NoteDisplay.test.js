import React from 'react';
import NoteDisplay from './NoteDisplay';
import { shallow } from 'enzyme';

let mockId = "ZN9zKmsle"
let list = [
  { id: "BqkLJDU45", isComplete: false, text: "Item 2" },
  { id: "Not Complets", isComplete: true, text: "Item 3" }
]
let mockTitle = "Another 2";

describe('NoteDisplay', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <NoteDisplay  id={mockId}
                    list={list}
                    title={mockTitle} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot', () => {
    const mockLongTitle = "THis is going to be a really long long title and this longer than than the max";
    const mockLongList = [
      { id: "BqkLJDU45", isComplete: false, text: "THis is going to be a really long long title and this longer than than the max" },
      { id: "Not Complets", isComplete: true, text: "Item 3" }
    ]
    wrapper = shallow(<NoteDisplay id={mockId} title={mockLongTitle} list={mockLongList} />)
  })

})
