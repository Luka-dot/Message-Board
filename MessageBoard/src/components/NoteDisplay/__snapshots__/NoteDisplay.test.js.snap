// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteDisplay should match the snapshot 1`] = `
<div
  className="NoteDisplay"
>
  <h3>
    Another 2
  </h3>
  <ul>
    <li
      className=""
      key="BqkLJDU45"
    >
      Item 2
    </li>
    <li
      className="completed"
      key="Not Complets"
    >
      Item 3
    </li>
  </ul>
</div>
`;
